packages/ui/src/theme/tokens/colors.ts

```diff
@@ -1,67 +1,87 @@
+import { DesignToken, WebDesignToken, ColorValue } from './types/designToken';
 import { OrdinalScale, OrdinalVariation } from './types/scales';
 
-export interface ColorScale {
-  10: DesignToken;
-  20: DesignToken;
-  40: DesignToken;
-  60: DesignToken;
-  80: DesignToken;
-  90: DesignToken;
-  100: DesignToken;
-}
-
-export interface FontColors extends OrdinalScale, OrdinalVariation {
-  inverse: DesignToken;
-  interactive: DesignToken;
-  hover: DesignToken;
-  focus: DesignToken;
-  active: DesignToken;
-  disabled: DesignToken;
-}
-
-export interface BackgroundColors extends OrdinalScale, OrdinalVariation {
-  disabled: DesignToken;
-}
-
-export interface BorderColors extends OrdinalScale {
-  disabled: DesignToken;
-  focus: DesignToken;
-  error: DesignToken;
-}
+type ScaleKeys = 10 | 20 | 40 | 60 | 80 | 90 | 100;
+
+type FontColors = {
+  inverse: DesignToken<ColorValue>;
+  interactive: DesignToken<ColorValue>;
+  hover: DesignToken<ColorValue>;
+  focus: DesignToken<ColorValue>;
+  active: DesignToken<ColorValue>;
+  disabled: DesignToken<ColorValue>;
+} & OrdinalScale &
+  OrdinalVariation;
+
+type BackgroundColors = {
+  disabled: DesignToken<ColorValue>;
+} & OrdinalScale &
+  OrdinalVariation;
+
+type BorderColors = {
+  disabled: DesignToken<ColorValue>;
+  focus: DesignToken<ColorValue>;
+  error: DesignToken<ColorValue>;
+} & OrdinalScale;
 
 type ColorTypes =
-  | ColorScale
+  | { [key in ScaleKeys]: DesignToken<ColorValue> }
   | FontColors
   | BackgroundColors
-  | DesignToken
+  | DesignToken<ColorValue>
   | BorderColors;
 
-export interface Colors {
+export type Colors = {
   // base color palette
-  red: ColorScale;
-  orange: ColorScale;
-  yellow: ColorScale;
-  green: ColorScale;
-  teal: ColorScale;
-  blue: ColorScale;
-  purple: ColorScale;
-  pink: ColorScale;
-  neutral: ColorScale;
-  white: DesignToken;
-  black: DesignToken;
+  red: { [key in ScaleKeys]: DesignToken<ColorValue> };
+  orange: { [key in ScaleKeys]: DesignToken<ColorValue> };
+  yellow: { [key in ScaleKeys]: DesignToken<ColorValue> };
+  green: { [key in ScaleKeys]: DesignToken<ColorValue> };
+  teal: { [key in ScaleKeys]: DesignToken<ColorValue> };
+  blue: { [key in ScaleKeys]: DesignToken<ColorValue> };
+  purple: { [key in ScaleKeys]: DesignToken<ColorValue> };
+  pink: { [key in ScaleKeys]: DesignToken<ColorValue> };
+  neutral: { [key in ScaleKeys]: DesignToken<ColorValue> };
+  white: DesignToken<ColorValue>;
+  black: DesignToken<ColorValue>;
 
   // Semantic colors
   font: FontColors;
   background: BackgroundColors;
   border: BorderColors;
   brand: {
-    primary: ColorScale;
-    secondary: ColorScale;
+    primary: { [key in ScaleKeys]: DesignToken<ColorValue> };
+    secondary: { [key in ScaleKeys]: DesignToken<ColorValue> };
   };
 
   [key: string]: ColorTypes | Record<string, ColorTypes>;
-}
+};
+
+export type WebColors = {
+  red: { [key in ScaleKeys]: WebDesignToken<ColorValue> };
```

Comment:
What do you think about creating some special types for colors? like:

```ts
export type ColorToken = WebDesignToken<ColorValue>;
export type ColorTokenMap<Keys extends keyof any> = { [key in Keys]: ColorToken };
```

then, it can be referenced as:

```ts
export type WebColors = {
  red: ColorTokenMap<ScaleKeys>;
  orange: ColorTokenMap<ScaleKeys>;
  ...
  white: ColorToken;
  black: ColorToken;
  ...
  font: ColorTokenMap<keyof Colors['font']>;
  background: ColorTokenMap<keyof Colors['background']>;
  border: ColorTokenMap<keyof Colors['border']>;
...
}
```